test/controllers/api/v2/config_reports_controller_test.rb

```diff
@@ -74,7 +74,7 @@ def test_create_duplicate
 
       Resolv.any_instance.stubs(:getnames).returns(['else.where'])
       post :create, params: { :config_report => create_a_puppet_transaction_report }
-      assert_response :redirect
+      assert_response :forbidden
```

Comment:
This surprises me. I would expect it to enforce SSL and redirect now, but it's going the other direction? Perhaps the SSL middleware is not applied in tests?

That does mean the whole test case becomes unreliable. This should be unreachable code in an end-to-end test, right?

Reply:
I didn't question it as I thought I'm flipping it back, but I believe it simply does not work as you can't change the middlewares during the run so eighter it is on, or it is off. but:

1. not in controller tests where the rack is not used at all
2. probably not even in the integration tests, because the rack is initialized once, there you probably have at least chance to change the options of middleware

That being said, I do not know how to test for this.